### ISSUE TITLE
No context menu for files and folders in Nautilus (Gnome)

### ISSUE THREAD

**Issue by gmagic** [flathub/com.synology.SynologyDrive#2]
Hi!
Thanks for your flatpak package. It's very usefull while Synology stopped supporting rpm packages for linux.

Here's one main issue in flatpak that bother: no context menu for files and folders in Nautilus (Gnome). Is there any way to make SynologyDrive preferences work with linux DE? Maybe there can be some flatpak app prefs?

**Comment by HorstBaerbel** [comment]
Yeah, that would be nice. Here on Fedora 35 is see no overlay icons either

**Comment by smidm** [comment]
The nautilus extension is located in `usr/lib/nautilus/extensions-3.0/libnautilus-drive-extension.so` after extracting the deb package. I tried to copy the extension to `/usr/lib64/nautilus/extensions-3.0/` where nautilus plugins reside on Fedora, but to no avail. This method works on Fedora as discussed in https://community.synology.com/enu/forum/1/post/121686?page=4&reply=469197
Packaging nautilus extension as Flatpak is probably possible only for Flatpak packaged nautilus / gnome-files. 
It would be nice to keep the libnautilus-drive-extension.so in the Flatpak and figure out a way to enable it in specific distros by copying / linking it somewhere manually.

**Comment by RayJW** [comment]
It seems like this is not possible at the moment (see [this](https://github.com/flatpak/flatpak/issues/4323) Flatpak issue). I guess we keep this open for now, as manually copying is a workaround until there is a solution for this.

**Comment by smidm** [comment]
The extension actually causes crashes, so it's not worth of it, see https://github.com/EmixamPP/synology-drive/issues/5#issuecomment-2044162373

**Comment by RayJW** [comment]
The Flatpak people closed this as not planned, seems like the way Synology Drive currently handles the Nautilus integration can't be done safely from within Flatpak. I'll close this as not planned.
